### PR TITLE
chore: disable persistent journaling feature

### PR DESCRIPTION
**Pull request by romange**
1. We have not worked on it for many months
2. It's not on the short term roadmap
3. It complicates the code around the replication.

### CODE REVIEW

**Comment by yuelimv** [comment]
`#include "core/uring.h"` from `journal_slice.h` can be further removed.

File: src/server/journal/journal_slice.cc [hunk 1]
```diff
@@ -111,9 +114,11 @@ error_code JournalSlice::Close() {
 
   return ec;
 }
+#endif
 
 void JournalSlice::AddLogRecord(const Entry& entry, bool await) {
   DCHECK(ring_buffer_);
+#if 0
```
Comment (adiholden) [inline src/server/journal/journal_slice.cc:121]: I believe we need the lsn++ operation inside the commented out code.
Did you run the replication regression tests and they passed?

File: src/server/journal/journal_slice.h [hunk 2]
```diff
@@ -53,8 +54,8 @@ class JournalSlice {
  private:
   struct RingItem;
 
-  std::string shard_path_;
-  std::unique_ptr<LinuxFile> shard_file_;
+  // std::string shard_path_;
+  // std::unique_ptr<LinuxFile> shard_file_;
   std::optional<base::RingBuffer<RingItem>> ring_buffer_;
```
Comment (adiholden) [inline src/server/journal/journal_slice.h:59]: why do we need ring buffer?

File: src/server/journal/journal_slice.cc (hunk 1 above)
Comment (romange) [reply src/server/journal/journal_slice.cc]: fixed

**Review by adiholden — APPROVED** [review]

**Pull request merged**